### PR TITLE
feat: add --cherry-pick-options to add to all cherry-pick run

### PR DESCRIPTION
**Pull request by earl-warren**
Fixes: https://github.com/kiegroup/git-backporting/issues/111

### CODE REVIEW

File: src/service/git/git-cli.ts [hunk 1]
```diff
@@ -110,10 +110,14 @@ export default class GitCLIService {
...
    * @param sha commit sha
    */
-  async cherryPick(cwd: string, sha: string, strategy = "recursive", strategyOption = "theirs"): Promise<void> {
+  async cherryPick(cwd: string, sha: string, strategy = "recursive", strategyOption = "theirs", cherryPickOptions: string | undefined): Promise<void> {
     this.logger.info(`Cherry picking ${sha}`);
     
-    const options = ["cherry-pick", "-m", "1", `--strategy=${strategy}`, `--strategy-option=${strategyOption}`, sha];
+    let options = ["cherry-pick", "-m", "1", `--strategy=${strategy}`, `--strategy-option=${strategyOption}`];
+    if (cherryPickOptions !== undefined) {
+      options.concat(cherryPickOptions.split(" "));
```
Comment (lampajr) [inline src/service/git/git-cli.ts:118]:
```suggestion
      options = options.concat(cherryPickOptions.split(" "));
```

**Review by lampajr — CHANGES_REQUESTED** [review]
Hi @earl-warren, overall the PR looks great!!

There is just one required change as the `options.concat` method _"returns a new array without modifying any existing arrays."_

File: src/service/git/git-cli.ts (hunk 1 above)
Comment (earl-warren) [reply src/service/git/git-cli.ts]: :facepalm: 

Thanks for catching this, I was debugging it manually and you fixed it!

**Review by lampajr — APPROVED** [review]

**Comment by lampajr** [comment]
Thanks @earl-warren :heart:

**Pull request merged**